pkg/resource/deploy/step_generator.go

```diff
@@ -604,7 +604,7 @@ func (sg *stepGenerator) generateStepsFromDiff(
 			// If this resource is protected we can't replace it because that entails a delete
 			// Note that we do allow unprotecting and replacing to happen in a single update
 			// cycle, we don't look at old.Protect here.
-			if goal.Protect {
```

Comment:
Why change between `goal.Protect` and `new.Protect`? Will they always agree?

Reply:
Yeh we set new.Protect from goal.Protect higher up in the step generator.